src/target/stm32f1.c

```diff
@@ -151,7 +160,21 @@ bool gd32f1_probe(target_s *target)
 		else
 			target->driver = "GD32F1";
 		break;
+	case 0x444U: /* Gigadevice gd32e50x */
+		target->driver = "GD32E5";
+		block_size = 0x2000;
+		break;
 	default:
+#if 0
+		/* If stm32f1_read_idcode() is not patchable for M33, then we'll do it here */
+		if ((target->cpuid & CPUID_PARTNO_MASK) == CORTEX_M33) {
+			const uint16_t device_id2 = target_mem_read32(target, DBGMCU_IDCODE_GD32E5) & 0xfffU;
+			if (device_id2 == 0x444U) {
+				target->driver = "GD32E5";
+				break;
+			}
+		}
+#endif
```

Comment:
Please remove this `#if 0` in favour of the change to the ID Code read function

Reply:
Okay. This was my initial version, as I wasn't sure if I was allowed to touch generic `stm32f1_read_idcode()`. I cannot test breakage for other targets this might introduce, however, the only M33's I know are not using ancient stm32f1 flash peripherals.

Reply:
That's absolutely fine, We'd prefer to build a set of comprehensive rules in stm32f1_read_idcode() which we refactored out for this exact cause and reason, than having ad-hoc readouts of the DBGMCU ID Code information